diff -ur zabbix-3.2.3-orig/src/zabbix_java/src/com/zabbix/gateway/JMXItemChecker.java zabbix-3.2.3/src/zabbix_java/src/com/zabbix/gateway/JMXItemChecker.java
--- zabbix-3.2.3-orig/src/zabbix_java/src/com/zabbix/gateway/JMXItemChecker.java	2016-12-21 09:08:40.000000000 +0100
+++ zabbix-3.2.3/src/zabbix_java/src/com/zabbix/gateway/JMXItemChecker.java	2017-03-23 15:23:49.438091735 +0100
@@ -55,7 +55,11 @@
 			String conn = request.getString(JSON_TAG_CONN);
 			int port = request.getInt(JSON_TAG_PORT);
 
-			url = new JMXServiceURL("service:jmx:rmi:///jndi/rmi://[" + conn + "]:" + port + "/jmxrmi");
+			if (conn.startsWith("_")) {
+				url = new JMXServiceURL(conn.replaceAll("_([^.]+).", "$1:").replaceFirst("(.*):", "$1://").replaceAll("__", "+") + ":" + port);
+			} else {
+				url = new JMXServiceURL("service:jmx:rmi:///jndi/rmi://[" + conn + "]:" + port + "/jmxrmi");
+			}
 			jmxc = null;
 			mbsc = null;
 
